core/txpool/subpool.go

```diff
@@ -35,14 +36,139 @@ type LazyTransaction struct {
 	Hash common.Hash        // Transaction hash to pull up if needed
 	Tx   *types.Transaction // Transaction if already resolved
 
-	Time      time.Time    // Time when the transaction was first seen
-	GasFeeCap *uint256.Int // Maximum fee per gas the transaction may consume
-	GasTipCap *uint256.Int // Maximum miner tip per gas the transaction can pay
+	Time time.Time // Time when the transaction was first seen
+	Fees uint256.Int
 
 	Gas     uint64 // Amount of gas required by the transaction
 	BlobGas uint64 // Amount of blob gas required by the transaction
 }
 
+type TxFees struct {
+	From common.Address // sender
+	Fees uint256.Int    // miner-fees earned by this transaction.
+}
+
+type FeeList []*TxFees
+
+func (f FeeList) Len() int {
+	return len(f)
+}
+
+func (f FeeList) Less(i, j int) bool {
+	return f[i].Fees.Lt(&f[j].Fees)
+}
+
+func (f FeeList) Swap(i, j int) {
+	f[i], f[j] = f[j], f[i]
+}
+
+func (f *FeeList) Push(x any) {
+	*f = append(*f, x.(*TxFees))
+}
+
+func (f *FeeList) Pop() any {
+	old := *f
+	n := len(old)
+	x := old[n-1]
+	old[n-1] = nil
+	*f = old[0 : n-1]
+	return x
+}
+
+type PendingSet struct {
```

Comment:
```suggestion
type pendingSet struct {
```
maybe. The exists an interface we can use